### PR TITLE
Dm 2550

### PR DESCRIPTION
**Pull request by JoshingYou1**
### JIRA issue link
https://agile6.atlassian.net/browse/DM-2550

## Description - what does this code do?
1. Addresses mobile/tablet/desktop design issues.
2. Potentially helps to improve loading times for `Diffusion Map`.
3. Other QOL code cleanup.

## Testing done - how did you test it/steps on how can another person can test it 
Make sure the fixes highlighted in green in this google doc [](https://docs.google.com/spreadsheets/d/15aa1z6yQV9GL6SOqxYrWd-gul8dkzw2UWYW9WDKuczA/edit#gid=0) have been accounted for.

## Acceptance criteria
- We can talk about how to break this out or we can timebox 
- Includes reworked designs from mobile review

## Definition of done
- [ ] Unit tests written (if applicable)
- [ ] e2e/accessibility tests written (if applicable)
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [X] A link has been provided to the originating JIRA issue
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: app/views/page/show.html.erb [hunk 1]
```diff
@@ -116,21 +116,21 @@
...
 
-        <%# Subpage Hyperlink %>
+          <%# Subpage Hyperlink %>
         <% when 'PageSubpageHyperlinkComponent' %>
-          <div class="grid-row margin-bottom-3  grid-col-8 <%= 'margin-x-auto' if @page.narrow? %>">
-            <a href="<%= component.url %>" class="width-100 dm-internal-link">
-              <h2 class="font-sans-xl display-inline x0-top"><%= component.title %></h2>
+          <div class="grid-row margin-bottom-5 grid-col-12 desktop:grid-col-8 <%= 'margin-x-auto' if @page.narrow? %>">
+            <a href="<%= component.url %>" class="width-100 dm-internal-link subpage-hyperlink-component">
+              <h2 class="font-sans-xl display-inline x0-top text-underline"><%= component.title %></h2>
```
Comment (emyl3) [inline app/views/page/show.html.erb:123]: can we remove the `text-underline` here and then get rid of the `subpage-hyperlink-component` class?

File: app/assets/stylesheets/dm/components/_links.scss [hunk 2]
```diff
@@ -70,3 +70,7 @@ a {
     }
   }
 }
+
+.subpage-hyperlink-component {
```
Comment (emyl3) [inline app/assets/stylesheets/dm/components/_links.scss:74]: I think this can be removed based on my comment in this file `app/views/page/show.html.erb`

**Review by emyl3 — CHANGES_REQUESTED** [review]
Great work on this Josh! Left some comments and also noticed that the bookmark button for the practice card in the "Highlighted practice" section is off

Screen Shot 2021-06-14 at 9 53 27 AM

File: app/views/page/show.html.erb (hunk 1 above)
Comment (JoshingYou1) [reply app/views/page/show.html.erb:123]: If I do that, then the extra line shows up at the end of the text again. 
Screen Shot 2021-06-14 at 1 19 57 PM
Screen Shot 2021-06-14 at 1 20 07 PM

**Comment by JoshingYou1** [comment]
> Great work on this Josh! Left some comments and also noticed that the bookmark button for the practice card in the "Highlighted practice" section is off
> 
> Screen Shot 2021-06-14 at 9 53 27 AM

Fixed!! Good catch!

File: app/views/page/show.html.erb (hunk 1 above)
Comment (emyl3) [reply app/views/page/show.html.erb:123]: Oh I didn't see that. Good call. You can ignore this then :)

File: app/assets/stylesheets/dm/components/_links.scss (hunk 2 above)
Comment (emyl3) [reply app/assets/stylesheets/dm/components/_links.scss:74]: please ignore

File: app/views/page/show.html.erb (hunk 1 above)
Comment (JoshingYou1) [reply app/views/page/show.html.erb:123]: All right, sounds good :blush:

File: app/assets/stylesheets/dm/components/_links.scss (hunk 2 above)
Comment (JoshingYou1) [reply app/assets/stylesheets/dm/components/_links.scss:74]: No worries :relaxed:

**Review by emyl3 — APPROVED** [review]
Looks good to me! 👍

**Review by bradjohnson92008 — APPROVED** [review]
Thx Josh - Great job!

**Pull request merged**